Essentials/src/main/java/com/earth2me/essentials/commands/Commandwarpinfo.java

```diff
@@ -0,0 +1,85 @@
+package com.earth2me.essentials.commands;
+
+import com.earth2me.essentials.CommandSource;
+import com.earth2me.essentials.User;
+import net.ess3.api.IUser;
+import org.bukkit.Location;
+import org.bukkit.Server;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.earth2me.essentials.I18n.tl;
+
+public class Commandwarpinfo extends EssentialsCommand {
+
+    public Commandwarpinfo() {
+        super("warpinfo");
+    }
+
+    @Override
+    public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
+        if (args.length == 0) {
+            if (!user.isAuthorized("essentials.warpinfo")) {
```

Comment:
Why is perm only checked if there are no args? Seems like a flaw to me

Reply:
The command framework already checks `essentials.warpinfo`, this is redundant